mlir-tensorrt/tensorrt/include/mlir-tensorrt-dialect/TensorRT/IR/TensorRTOps.td

```diff
@@ -71,7 +71,9 @@ def TensorRT_TensorRTModuleOp : TensorRT_Op<"module", [
 
   let extraClassDeclaration = [{
     /// Return the name of this module if present.
-    std::optional<StringRef> getName() { return getSymName(); }
+    std::optional<StringRef> getName() const {
+      return (*this)->getAttrOfType<StringAttr>("sym_name").getValue();
```

Comment:
Why was this changed? Generally you don't want to rely on accessing properties through the string literal name.

Reply:
It was required because we were passing trtModule why ArrayRef which I think is equivalent to passing by `const &`.
Accessing methods such as `getSymName` require the calling object to be `non const`.

I have reverted this change and instead now passing trtModule as `SmallVectorImpl<tensorrt::TensorRTModuleOp>& trtModules)`